src/js_src/lib/searchHelpers.js

```diff
@@ -37,6 +37,11 @@ export function makeFieldDisplayName(unformattedName) {
   }
 }
 
+export function getFilterQueryParam(key, val, queryParams) {
```

Comment:
I think this new method is correctly and ingeniously used.  However, it might make sense to rename these two methods to intuitively describe how to use one or the other.  This new method gets the new query params explicitly removing any page num, so maybe something like `getQueryParamsNoPage`.